inc/class-core-sitemaps-posts.php

```diff
@@ -25,18 +25,58 @@ public function __construct() {
 	 * @noinspection PhpUnused
 	 */
 	public function render_sitemap() {
-		$sitemap = get_query_var( 'sitemap' );
-		$paged   = get_query_var( 'paged' );
+		global $wp_query;
 
+		$sitemap  = get_query_var( 'sitemap' );
+		$sub_type = get_query_var( 'sub_type' );
+		$paged    = get_query_var( 'paged' );
+
+		$sub_types = $this->get_object_sub_types();
+
+		if ( ! isset( $sub_types[ $sub_type ] ) ) {
```

Comment:
This causes every front-end page in WP to 404 :)

The check for subtype needs to happen after we've determined that `$this->slug === $sitemap`, otherwise, we shouldn't be doing anything.

Reply:
Wow can't believe I did that. 🤡 
Moved most of the logic within the context of the current sitemap.